pkg/steps/release/promote.go

```diff
@@ -173,6 +157,47 @@ func (s *promotionStep) Run(ctx context.Context, dry bool) error {
 	return nil
 }
 
+// ToPromote determines the mapping of local tag to external tag which should be promoted
+func ToPromote(config api.PromotionConfiguration, images []api.ProjectDirectoryImageBuildStepConfiguration, requiredImages sets.String) (map[string]string, sets.String) {
+	tagsByDst := make(map[string]string)
+	names := sets.NewString()
+
+	if config.Disabled {
+		return tagsByDst, names
+	}
+
+	for _, image := range images {
+		// if the image is required or non-optional, include it in promotion
+		tag := string(image.To)
+		if requiredImages.Has(tag) || !image.Optional {
+			tagsByDst[tag] = tag
+			names.Insert(tag)
+		}
+	}
+	for _, tag := range config.ExcludedImages {
+		delete(tagsByDst, tag)
+		names.Delete(tag)
+	}
+	for dst, src := range config.AdditionalImages {
+		tagsByDst[dst] = src
+		names.Insert(dst)
+	}
+
+	if config.NamePrefix == "" {
+		return tagsByDst, names
+	}
+
+	namesByDst := map[string]string{}
```

Comment:
nit: on L162, you `make`, here you `{}`  - make it consistent

Reply:
ah, other was preeexisting